### PR TITLE
added possibility to stop objects

### PR DESCRIPTION
**Pull request by HannesBachter**
objects that are moving stop when too close to a chosen object (-o --object)

works at the moment only for linear motion

(sorry for the 2 unnecessary commits, couldn't rebase)

### CODE REVIEW

File: cob_bringup_sim/scripts/move_object.py [hunk 1]
```diff
@@ -106,35 +121,38 @@ def move_on_line(self, start, goal):
...
-            object_new_pose = Pose()
-            object_new_pose.position.x = step_x
-            object_new_pose.position.y = step_y
-            quat = tf.transformations.quaternion_from_euler(0.0, 0.0, yaw)
-            object_new_pose.orientation.x = quat[0]
-            object_new_pose.orientation.y = quat[1]
-            object_new_pose.orientation.z = quat[2]
-            object_new_pose.orientation.w = quat[3]
+            model_dist = self.get_model_dist(step_x, step_y)
+            if(model_dist > 2):
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/move_object.py:129]: please introduce a parameter `stop_distance` with default = 2.

File: cob_bringup_sim/scripts/move_object.py [hunk 2]
```diff
@@ -92,6 +92,21 @@ def __init__(self):
...
+    def get_model_dist(self, x, y):
+        if self.options.model != None:
+            req = GetModelStateRequest()
+            req.model_name = self.options.model
+            rospy.wait_for_service('/gazebo/get_model_state')
+            client = rospy.ServiceProxy('/gazebo/get_model_state', GetModelState)
+            res = client(req)
+            model_dist = math.sqrt(math.pow(res.pose.position.x-x,2) + math.pow(res.pose.position.y-y,2))
+        else:
+            model_dist = 5;
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/move_object.py:106]: should be `model_dist = stop_distance`

File: cob_bringup_sim/scripts/move_object.py [hunk 3]
```diff
@@ -215,6 +233,11 @@ def parse_options(self):
         parser.add_option("-r", "--radius",
             dest="radius", default=None,
             help="Radius, only used for circular movement. Default: None")
+            
+        parser.add_option("-o", "--object",
```
Comment (floweisshardt) [inline cob_bringup_sim/scripts/move_object.py:237]: please rename to `stop_objects` (plural) as a list of strings and change code to check against all objects in the list.

**Review by floweisshardt — CHANGES_REQUESTED** [review]
see inline comments

File: cob_bringup_sim/scripts/move_object.py (hunk 2 above)
Comment (fmessmer) [reply cob_bringup_sim/scripts/move_object.py:106]: In the else case, the ModelState cannot be retrieved....I think in this case, the object.model is does not exist (?) and we do not want to stop....thus we would set the model_dist to something ```float('inf')```

**Comment by fmessmer** [comment]
I addressed @ipa-fmw's comments in #125, i.e.
 - introduce stop_distance (--stop-distance)
 - able to pass a list of objects (--stop-objects); -o/--object is not possible anymore
 - stop also on circle and polygon

@fmw-hb please test #125

**Pull request merged**